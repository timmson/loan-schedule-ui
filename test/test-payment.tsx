import React from "react"
import renderer from "react-test-renderer"
import Payment from "../src/payment"


describe("Payment should", () => {

	let component = null
	const payment = {
		paymentDate: "25.10.2012"
	}

	beforeAll(() => {
		component = renderer.create(<Payment payment={payment} onClick={(event) => event}/>)
	})

	test("trigger click", () => {
		const expected = 1

		const onClickResult =  component.root.findByType("tr").props.onClick(expected)

		expect(onClickResult).toEqual(expected)
	})

	test("equal to snapshot", () => {
		expect(component.toJSON()).toMatchSnapshot()
	})

	afterAll(() => component.unmount())

})