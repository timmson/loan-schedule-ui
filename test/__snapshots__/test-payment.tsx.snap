// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment should equal to snapshot 1`] = `
<tr
  className=""
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <td>
    25.10.2012
  </td>
  <td />
  <td
    className="desktop-cell"
  />
  <td
    className="desktop-cell"
  />
  <td />
  <td />
</tr>
`;
